node/src/tasks/bsp_delete_file.rs

```diff
@@ -0,0 +1,150 @@
+use anyhow::anyhow;
+use sc_tracing::tracing::*;
+use shc_blockchain_service::events::{BspConfirmStoppedStoring, FinalisedBspConfirmStoppedStoring};
+use shc_forest_manager::traits::ForestStorage;
+use sp_core::H256;
+
+use crate::services::handler::StorageHubHandler;
+use crate::tasks::{BspForestStorageHandlerT, FileStorageT, NoKey};
+use shc_actors_framework::event_bus::EventHandler;
+
+const LOG_TARGET: &str = "bsp-delete-file-task";
+
+pub struct BspDeleteFileTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    storage_hub_handler: StorageHubHandler<FL, FSH>,
+}
+
+impl<FL, FSH> Clone for BspDeleteFileTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    fn clone(&self) -> BspDeleteFileTask<FL, FSH> {
+        Self {
+            storage_hub_handler: self.storage_hub_handler.clone(),
+        }
+    }
+}
+
+impl<FL, FSH> BspDeleteFileTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    pub fn new(storage_hub_handler: StorageHubHandler<FL, FSH>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+
+    async fn remove_file_from_forest(&self, file_key: &H256) -> anyhow::Result<()> {
+        // Remove the file key from the Forest.
+        // Check that the new Forest root matches the one on-chain.
```

Comment:
I believe you don't check that the forest root matches the one on chain.

But since this task is async to removing the file from the forest in the runtime, there could be a scenario where there is multiple files being removed but this task is executed in a different order from the order on-chain. 

For example:
- on-chain > delete file 1
- on-chain > delete file 2
- task > remove file 1 from forest > fails check since forest root on chain is the result of removing both files
- task > remove file 2 from forest > succeeds

Reply:
I have removed the comment and indeed I don't check the Forest root.

I am working on an other PR for an integration test that test multiple file deletion for a BSP. I will check those scenarios.